### PR TITLE
Edit Data: Fix editing stopping after tab switch

### PR DESCRIPTION
**Pull request by benrr101**
A few changes here:
1) Upgrading the version of angular2-slickgrid to take a change that exposes the heavily utilized ActiveCellChanged event. Associated yarn lock updates
2) Updating the edit data grid to use the exposed ActiveCellChanged angular event instead of binding directly to slickgrid event -- this fixes issue where edit data stops working after switching tabs
3) Removing the commit from row revert. This shouldn't have been there to begin with.
4) Fixed undiscovered issue where pressing escape on the 2nd to last row always reverts the row and removes it. It should only do that if you're adding a new row.
5) Removing unused arguments for column from a couple private methods.

There's still a few issues with editing and switching tabs, but they will require bigger changes I don't want to deal with in this review.

### CODE REVIEW

File: build/yarn.lock [hunk 1]
```diff
@@ -585,7 +585,7 @@ extend@~1.2.1:
 "extensions-modules@file:../extensions-modules":
   version "0.1.0"
   dependencies:
-    dataprotocol-client "file:../../../../Library/Caches/Yarn/v1/dataprotocol-client"
+    dataprotocol-client "file:../../../Library/Caches/Yarn/v1/dataprotocol-client"
```
Comment (MattIrv) [inline build/yarn.lock]: what's this change? it doesn't look related to the slick grid update

Comment (anthonydresser) [reply build/yarn.lock]: This is a side affect of yarn. we will continue to have these changes until we move the dataprotocol client out and use either github reference of npm reference.

Comment (benrr101) [reply build/yarn.lock]: I'm guessing they're pretty much benign since I was able to build with the wrong path (for me) and travis can build with the wrong path (for it)

File: package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "@angular/router": "~4.1.3",
     "@angular/upgrade": "~4.1.3",
     "angular2-grid": "2.0.6",
-    "angular2-slickgrid": "git://github.com/Microsoft/angular2-slickgrid.git#1.3.7",
+    "angular2-slickgrid": "git://github.com/Microsoft/angular2-slickgrid.git#1.3.8",
```
Comment (kburtram) [inline package.json]: the most recent vbump for this module broke the restore dialog.  please verify the restore dialog still works with this change.

**Comment by kburtram** [comment]
@benrr101 I'll merge this PR for you so it gets included in the Feb test passed with the build I'm producing tonight.
